### DISCUSSION TITLE
Table 部分 API 讨论

### DISCUSSION THREAD

**Discussion by WynterDing** [ant-design/pro-components#63] [General]
目前有一些 Table 有一些 API 是不是可以稍微调整一下, 大家一起 看下

1. 对 request 请求返回的参数进行处理现在叫 `postData`，感觉叫 `dataFormatter` 更合理
2. beforeSearchSubmit 是不是可以放到 search API 的结构 里面，改为 `search.beforeSubmit`
3. 废弃 type 、Columns 中的 hideInForm 这两个 API， 从现在 pro-components 的计划来看，不再需要这些 API 了

**Comment by afc163** [comment]
+1

**Comment by yutingzhao1991** [comment]
+1

**Comment by kaoding** [comment]
这是我之前针对 ProTable API 提的一些需求
https://github.com/ant-design/pro-table/issues/145

**Comment by WynterDing** [comment]
> 这是我之前针对 ProTable API 提的一些需求
> [ant-design/pro-table#145](https://github.com/ant-design/pro-table/issues/145)

你提的这些，最后都会体现在这个仓库的组件上，比如 Form 会单独独立出来，而 Search 部分，不会从 Table 中导出，而是最后作为 FilterForm 导出给你自定义使用。Table 还是会聚焦在 表格的 开箱即用、功能预设 这两大原则上。

**Comment by kaoding** [comment]
> > 这是我之前针对 ProTable API 提的一些需求
> > [ant-design/pro-table#145](https://github.com/ant-design/pro-table/issues/145)
> 
> 你提的这些，最后都会体现在这个仓库的组件上，比如 Form 会单独独立出来，而 Search 部分，不会从 Table 中导出，而是最后作为 FilterForm 导出给你自定义使用。Table 还是会聚焦在 表格的 开箱即用、功能预设 这两大原则上。

我这个是提的比较早的，大部分以不同的方式实现了，只是为讨论 AP I设计时提供一些资料，谢谢！

**Comment by mtianyan** [comment]
希望ProForm 可以有方式和Protable 中 type="form“ 传入columns 就可以使用一样方便。 目前我是使用columns + type="form"享受自动表单的便利， 然后optionRender =False，自己再添加一些额外的columns 之外的自定义表单， 最终通过formRef 收集所有数据，加上自己的数据自行发送。这样实现增改需求非常简洁，通用，不希望这个被拿掉，如果ProForm 作为type="form“ 的Protable 的替代品，希望可以有一致的columns传入形成整个表单的预设使用方式。

**Comment by mtianyan** [comment]
看了下ProForm 暂时还未稳定，hideInForm 被用在type="form“ 根据columns 自动形成新增修改表单时是很有用的，不建议拿掉呢。

**Comment by rojer95** [comment]
valueType能否在table和搜索的form中单独设置呢？

比如有一个订单时间在table中是具体的时间，为了让table自动格式化时间所以用的valueType是dateTime，
但是搜索的时候是希望能够搜索某个时间范围的订单，所以要用的valueType是dateTimeRange

但是目前好像不能单独设置valueType
虽然可以通过renderText来单独渲染table的字段，
但是我个人觉得 搜索某个时间范围 的这种场景会比较多，
那么这就不怎么“开箱即用”了

**Comment by chenshuai2144** [comment]
valueType 支持传入一个 function 的，你可以试试

**Comment by aaronxdd** [comment]
能否提供单独重置某个search param 的API

**Comment by chenshuai2144** [comment]
@aaronxdd  form.setFields
